### PR TITLE
chore: Update `miette`

### PR DESCRIPTION
**Pull request by Brooooooklyn**

### CODE REVIEW

**Review by kdy1 — COMMENTED** [review]
swc-bump:
 - swc_error_reporters
 - swc_core

**Review by kdy1 — APPROVED** [review]

**Comment by kdy1** [comment]
I'll update the test references and merge the PR.

File: crates/swc_error_reporters/src/lib.rs [hunk 1]
```diff
@@ -266,8 +266,8 @@ impl fmt::Display for MietteDiagnostic<'_> {
 
 fn convert_span(span: Span) -> SourceSpan {
     let len = span.hi - span.lo;
-    let start = SourceOffset::from(span.lo.0 as usize);
-    SourceSpan::new(start, SourceOffset::from(len.0 as usize))
+    let start = SourceOffset::from(span.lo.0 as usize + 1);
```
Comment (kdy1) [inline crates/swc_error_reporters/src/lib.rs:269]: Is this correct?

Comment (Brooooooklyn) [reply crates/swc_error_reporters/src/lib.rs:269]: seems like should `+1` here:

image

image

**Review by kdy1 — DISMISSED** [review]

**Review by kdy1 — APPROVED** [review]

**Pull request merged**